### PR TITLE
Sign up

### PR DESCRIPTION
**Pull request by ishaque-mobikasa**
Added signup page, used shared pref. enabled local sign in and signup using shared preferences.
used getX package for state management, Routed to home screen using logged in status..
Used get Snack bar to inform the user in the sign up or login page errors..
Now obscuring button keeps the password visible for only 3 seconds..

### CODE REVIEW

File: lib/core/preferences.dart [hunk 1]
```diff
@@ -0,0 +1 @@
+ const String isLoggedIn="isLoggedIn";
```
Comment (tusharmobikasa) [inline lib/core/preferences.dart]: Please make a class of this and declare this in string file not in in core under utils make a class of all string and put there and Please read about the why to make class .

File: lib/presentation/pages/login/controller/login_controller.dart [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+  final RxBool isEnabled = false.obs;
+  onLoginButtonPress() {
+    if (loginKey.value.currentState!.validate()) {
+      String? dataOndisk = preferences.getString(emailController.value.text);
+      if (dataOndisk != null) {
+        UserModel user = UserModel.fromJson(jsonDecode(dataOndisk));
+        if (user.email == emailController.value.text &&
+            user.password == passwordController.value.text) {
+          preferences.setBool(isLoggedIn, true);
+          Get.snackbar("Valid Credentials", "You are Successfully Logged in");
```
Comment (tusharmobikasa) [inline lib/presentation/pages/login/controller/login_controller.dart]: if make string file declare all string over there

File: lib/core/validation.dart [hunk 3]
```diff
@@ -4,22 +4,15 @@ extension extString on String {
     return emailRegExp.hasMatch(this);
   }
 
-  bool get isValidName{
-    final nameRegExp =  RegExp(r"^\s*([A-Za-z]{1,}([\.,] |[-']| ))+[A-Za-z]+\.?\s*$");
```
Comment (tusharmobikasa) [inline lib/core/validation.dart:8]: format this code

File: lib/presentation/pages/login/controller/login_controller.dart [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+import 'dart:convert';
+
+import 'package:flutter/cupertino.dart';
+import 'package:get/get.dart';
+import 'package:shared_preferences/shared_preferences.dart';
+import 'package:task/core/preferences.dart';
+import 'package:task/core/routes.dart';
+import 'package:task/data/models/onBoard/user/user_model.dart';
+
+class LoginController extends GetxController {
```
Comment (tusharmobikasa) [inline lib/presentation/pages/login/controller/login_controller.dart:12]: First declare the variable the  use them 
why oninit is coming first and then all variables are declaring

File: lib/presentation/pages/sign_up/controller/sign_up_controller.dart [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+import 'dart:convert';
+
+import 'package:flutter/material.dart';
+import 'package:get/get.dart';
+import 'package:shared_preferences/shared_preferences.dart';
+import 'package:task/core/routes.dart';
+import 'package:task/data/models/onBoard/user/user_model.dart';
+
+class SignUpcontroller extends GetxController {
+  @override
```
Comment (tusharmobikasa) [inline lib/presentation/pages/sign_up/controller/sign_up_controller.dart]: Make variables first and the use it

File: pubspec.lock [hunk 6]
```diff
@@ -50,6 +50,13 @@ packages:
       url: "https://pub.dartlang.org"
     source: hosted
     version: "1.0.5"
+  dio:
+    dependency: "direct main"
```
Comment (tusharmobikasa) [inline pubspec.lock:54]: Why this file is again in git commit

File: macos/Flutter/GeneratedPluginRegistrant.swift [hunk 7]
```diff
@@ -6,7 +6,9 @@ import FlutterMacOS
 import Foundation
```
Comment (tusharmobikasa) [inline macos/Flutter/GeneratedPluginRegistrant.swift:6]: why this file in git commit

**Review by tusharmobikasa — COMMENTED** [review]
Please make a habbit of making classes and if using string class use that variable and also use dimens file to use padding and margins and try to not put all unwanted files in git also try to resolve conflict don't delete the file and make it again .

File: lib/presentation/pages/login/controller/login_controller.dart (hunk 4 above)
Comment (ishaque-mobikasa) [reply lib/presentation/pages/login/controller/login_controller.dart:12]: OK

File: lib/core/validation.dart (hunk 3 above)
Comment (ishaque-mobikasa) [reply lib/core/validation.dart:8]: it is Formatted already

File: lib/presentation/pages/login/controller/login_controller.dart (hunk 2 above)
Comment (ishaque-mobikasa) [reply lib/presentation/pages/login/controller/login_controller.dart]: Made a string file

File: lib/core/preferences.dart (hunk 1 above)
Comment (ishaque-mobikasa) [reply lib/core/preferences.dart]: ok

**Review by tusharmobikasa — APPROVED** [review]

**Pull request merged**